platform/lang-impl/src/com/intellij/codeInspection/ex/DescriptorComposer.java

```diff
@@ -130,7 +131,8 @@ protected void composeDescription(@NotNull CommonProblemDescriptor description,
     VirtualFile vFile = null;
 
     if (expression != null) {
-      vFile = expression.getContainingFile().getVirtualFile();
+      vFile = (expression instanceof PsiFileSystemItem ? (PsiFileSystemItem) expression : expression.getContainingFile()).getVirtualFile();
```

Comment:
Now can be written as `vFile = (expression instanceof PsiFileSystemItem item ? item : expression.getContainingFile()).getVirtualFile();`